src/controllers/AuthController.ts

```diff
@@ -106,264 +104,122 @@ class AuthController {
 		next: NextFunction
 	) {
-		// try {
-		// 	let {hash, otp, password}: ResetPasswordPayload = req.body
-		// 	// encrypt password
-		// 	const isValidPassword: boolean =
-		// 		await helper.regexPassword(password)
-		// 	if (!isValidPassword) {
-		// 		throw new BadRequestException(
-		// 			"Password must be more then 8 char!",
-		// 			"validation_error"
-		// 		)
-		// 	}
-		// 	password = await bcrypt.hash(
-		// 		password,
-		// 		parseInt(process.env.SALT_ROUNDS as string)
-		// 	)
-		// 	const decryptData: DecryptData = await decryptBycrypto(hash)
-		// 	const {email, userId} = decryptData
-		// 	// check if user & verification exist
-		// 	const [[authCredential], [verificationData]]: [
-		// 		CredentialDetails[],
-		// 		VerificationTableData[]
-		// 	] = await Promise.all([
-		// 		// authCredential
-		// 		this.authCredentialModel.list({userName: email}),
-		// 		// verification
-		// 		this.verificationCommonModel.list({
-		// 			value: email,
-		// 			isVerified: false,
-		// 			verificationType: LogInWith.EMAIL,
-		// 			verificationFor: VerificationFor.AUTH,
-		// 			status: true
-		// 		})
-		// 	])
-		// 	if (!authCredential) {
-		// 		throw new UnauthorizedException("User not found")
-		// 	}
-		// 	if (!verificationData) {
-		// 		throw new UnauthorizedException(
-		// 			"Invalid OTP. Please resend and try again."
-		// 		)
-		// 	}
-		// 	// check if otp is valid
-		// 	let decoded: any = null
-		// 	try {
-		// 		decoded = jwt.verify(
-		// 			(verificationData?.otp).toString(),
-		// 			process.env.JWT_SECRET_KEY as string
-		// 		)
-		// 	} catch (error) {
-		// 		throw new UnauthorizedException(
-		// 			"Invalid OTP. Please resend and try again."
-		// 		)
-		// 	}
-		// 	if (parseInt(decoded?.otp) !== parseInt(otp.toString())) {
-		// 		throw new UnauthorizedException(
-		// 			"Invalid OTP. Please resend and try again."
-		// 		)
-		// 	}
-		// 	if (
-		// 		new Date(
-		// 			new Date(verificationData.createdAt.toString()).getTime() +
-		// 				parseInt(
-		// 					process.env.OTP_EXPIRATION_IN_MINUTES as string
-		// 				) *
-		// 					60000
-		// 		).getTime() < new Date().getTime()
-		// 	) {
-		// 		throw new UnauthorizedException(
-		// 			"OTP expired. Please resend and try again."
-		// 		)
-		// 	}
-		// 	await Promise.all([
-		// 		// update otp
-		// 		this.verificationCommonModel.update(
-		// 			{isVerified: true},
-		// 			verificationData.verificationId
-		// 		),
-		// 		// update password
-		// 		this.authCredentialModel.update(
-		// 			{password},
-		// 			authCredential.credentialId
-		// 		)
-		// 	])
-		// 	return res.json({
-		// 		success: true,
-		// 		message: `Password updated successfully`
-		// 	})
-		// } catch (error) {
-		// 	res.status(400).json({
-		// 		status: 400,
-		// 		message: error?.toString(),
-		// 		code: "unexpected_error"
-		// 	})
-		// 	return
-		// }
+		try {
+			const response = new ApiResponse(res)
+			let {hash, otp, password}: ResetPasswordPayload = req.body
```

Comment:
Same as verify otp API.
use email instead of hash.

Reply:
done